hazelcast/src/main/java/com/hazelcast/map/impl/record/DataRecord.java

```diff
@@ -20,7 +20,7 @@
 
 class DataRecord extends AbstractRecord<Data> {
 
-    protected Data value;
+    protected volatile Data value;
```

Comment:
Could you add a line of comments on the volatiles? You can do this with a hard push; but then other people are not going to get confused and remove it.